modules/nat/main.tf

```diff
@@ -7,6 +14,11 @@ resource "azurerm_nat_gateway" "this" {
   tags = var.tags
 }
 
+resource "azurerm_subnet_nat_gateway_association" "this" {
+  subnet_id      = azurerm_subnet.this.id
+  nat_gateway_id = azurerm_nat_gateway.this.id
+}
```

Comment:
Pretty sure it's possible to create multiple associations, similar to `azurerm_nat_gateway_public_ip_association` below? 🤔

Reply:
Also we shouldn't hardcode a subnet inside this submodule, the ID of the subnets to associate should be passed in as variables.